### ISSUE TITLE
Flaky e2e tests on GKE

### ISSUE THREAD

**Issue by randmonkey** [Kong/kubernetes-ingress-controller#2574]
In the release actions for `v2.4.0`, the e2e tests on GKE passes for 1.23.5 and 1.20, but fails on 1.22:
- [release 334](https://github.com/Kong/kubernetes-ingress-controller/actions/runs/2499131274)
- [release 335](https://github.com/Kong/kubernetes-ingress-controller/actions/runs/2499337966)
I thought it was caused by timeout, so I changed the timeout from `30m` to `60m` in #2573, but the action still fails:
[release 336](https://github.com/Kong/kubernetes-ingress-controller/actions/runs/2500764853) The result is almost the same.

The tests passed on 1.23.5 and 1.20, seems to be no reason to fail on 1.22. So there might be some flaky tests or environment issues, or the update on timeout did not correctly applied in the actions.

**Comment by rainest** [comment]
Those failures were due to https://github.com/Kong/kubernetes-ingress-controller/pull/2578

The broader issue is that the multi-version tests only run for releases, not the regular nightly tests. We probably want to look at running those in nightlies, possibly even removing them from the release job (the final test run is mostly to ensure that we didn't screw something up with the final PR to create the release--we really shouldn't be introducing something that breaks integration compatibility in those) to speed up releases.

**Comment by mflendrich** [comment]
@randmonkey do you think that @rainest's comment that the issue is now resolved fixes the root problem? If so, please close the issue. :bow:

**Comment by randmonkey** [comment]
I think it needs some verification to prove that the issue is now resolved. I do not know how the failure of tests is related to Knative <1.22.

**Comment by rainest** [comment]
Verification is essentially that releases don't have the 1.22 GKE job fail anymore.

There's no good way to check the cause currently--I noticed that the logs we did have look weird because it was getting stuck on cluster startup and hitting the global timeout, so there must be some reason that version alone was not coming online. To get the specific cause I had to go check out the spawned cluster in the GKE console, where I could see the failing Pods and check their logs, and those had a clear "Knative don't work on this version no more" message.

Something similar to https://github.com/Kong/kubernetes-ingress-controller/blob/5baf53d1a4731782a798d669335a817d6c99b913/test/e2e/utils_test.go#L304-L322 could be used to get more detailed logs, though we'd want to port it to a KTF feature to use it in integration tests as well.

We don't currently make use of that since getting a full set of logs is quite noisy (especially in my original use case for it, where it ran during every tick of an `Eventually` within a test), but it should be possible to combine it with https://pkg.go.dev/github.com/stretchr/testify@v1.7.5/suite#TearDownTestSuite and some sort artifact upload instead of stdout to include container logs without making the controller logging a huge mess.

We may also want to consider a shorter timeout on cluster spawn itself, just so it doesn't take the full hour or whatever to report that it did nothing.
